tokio-util/src/codec/framed.rs

```diff
@@ -52,6 +52,53 @@ where
     /// calling [`split`] on the `Framed` returned by this method, which will
     /// break them into separate objects, allowing them to interact more easily.
     ///
+    /// Note that for some underlying byte stream sources, namely files and FIFOs,
+    /// it's possible to temporarily return 0 bytes and reach eof.
+    /// In those cases it is possible to resume the frame stream after
+    /// it as returned `None` by simply attempting to read from it again.
+    ///
+    /// Note that attempts to read from such an eof data source will not wait
+    /// for new data and simply return `None` if it's still eof.
+    ///
+    /// Receiving notifications for filesystem changes is very platform dependent
+    /// and out of scope for this implementation.
+    /// It is up to the caller to ensure rate limiting or implement a
+    /// notification system with the help of e.g.
+    /// inotify, FSEvents, FileSystemWatcher, IPC, or a [`watch`] between tasks.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use tokio::fs::OpenOptions;
+    /// use futures::sink::SinkExt;
+    /// use tokio_stream::StreamExt;
+    /// use tokio_util::codec::{LinesCodec, Framed};
+    /// # #[tokio::main]
+    /// # async fn main() -> std::io::Result<()> {
+    ///     let file = OpenOptions::new()
+    ///         .read(true)
+    ///         .append(true)
+    ///         .create(true)
+    ///         .open("/tmp/framed_append_file.txt")
+    ///         .await?;
+    ///
+    ///     let mut frames = Framed::new(file, LinesCodec::new());
+    ///
+    ///     while let Some(frame) = frames.next().await {
+    ///         //Do something with the frame...
+    ///     }
+    ///
+    ///     frames.send("Something that wasn't there before!").await.unwrap();
+    ///
+    ///     while let Some(frame) = frames.next().await {
+    ///         //Do something with the new frame...
+    ///     }
+    ///
+    /// # Ok(())
+    /// # }
+    /// ```
```

Comment:
I think this should probably just be taken out. This feature should not be taking up two thirds of the entire documentation for `Framed`. Consider instead augmenting the module documentation, which already has a more in-depth explanation.

Reply:
Will do!

Reply:
Some side snark: I'd probably decrease the ratio by adding more documentation to the rest of the methods, but that's personal taste 😆 🤷  

When I first used `framed`, an example would have helped, but it was also pretty much the first piece of tokio I've touched, so that might not be representative. Do you think the example could be useful as a more reduced/generic version (no resumed read) in the module documentation?

Reply:
Feel free to improve it! Note that I added some module docs in #3283, which may not be in your branch until your merge in master.

Reply:
I should have started using rust a week later, then I would have gotten your awesome docs.
I've removed the example and tried to condense the documentation of resuming to an absolute minimum.